kroxylicious-test-tools/src/test/java/io/kroxylicious/test/tester/KroxyliciousTestersTest.java

```diff
@@ -216,9 +216,15 @@ private static Map<String, Object> randomGroupIdAndEarliestReset() {
     }
 
     private static void assertOneRecordConsumedFrom(Consumer<String, String> consumer) {
-        consumer.subscribe(List.of(TOPIC));
-        ConsumerRecords<String, String> records = consumer.poll(Duration.ofSeconds(10));
-        assertEquals(1, records.count());
+        try {
+            consumer.subscribe(List.of(TOPIC));
+            ConsumerRecords<String, String> records = consumer.poll(Duration.ofSeconds(10));
+            assertEquals(1, records.count());
+        }
+        finally {
+            consumer.close();
```

Comment:
A bit of a surprising side effect of calling `assertOneRecordConsumedFrom`. 

Also isn't it considered bad form to close a resource someone else allocated?

Reply:
Reworked it a little to create-close the consumer with a try-with-resources, the test was also @Disabled with a comment about it not working from Keith, was this resolved with the changes around flushing @k-wall? The test passes now at least on my machine.

Reply:
@robobario "was this resolved with the changes around flushing" yes.  I'd forgotten - thank you.
@SamBarker assertOneRecordConsumedFrom - "close a resource someone else allocated" <- dumb change on my behalf. @robobario's refactor looks sensible!